cluster/query_executor.go

```diff
@@ -481,42 +482,29 @@ func (e *QueryExecutor) executeSelectStatement(stmt *influxql.SelectStatement, c
 	}
 
 	// Emit rows to the results channel.
-	var writeN int64
-	var emitted bool
-	for {
-		row := em.Emit()
-		if row == nil {
-			// Check if the query was interrupted while emitting.
-			select {
-			case <-closing:
-				return influxql.ErrQueryInterrupted
-			default:
-			}
-			break
-		}
-
-		result := &influxql.Result{
-			StatementID: statementID,
-			Series:      []*models.Row{row},
-		}
+	var emstats emitterStats
+	emitterCh := make(chan emitterStats, 1)
+	done := make(chan struct{})
+	go e.monitorEmitter(em, stmt, statementID, emitterCh, done, results, closing)
+	defer func() { <-done }()
```

Comment:
I had to move the emitting of points to a separate goroutine so that we can monitor stats separately.